packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -845,24 +1027,23 @@ class IOSDeviceLogReader extends DeviceLogReader {
   /// is true.
   final List<String> _streamFlutterMessages = <String>[];
 
-  /// When using both `idevicesyslog` and `ios-deploy`, exclude logs with the
-  /// "flutter:" prefix if they have already been added to the stream. This is
-  /// to prevent duplicates from being printed.
-  ///
-  /// If a message does not have the prefix, exclude it if the message's
-  /// source is `idevicesyslog`. This is done because `ios-deploy` and
-  /// `idevicesyslog` often have different prefixes on non-flutter messages
-  /// and are often not critical for CI tests.
+  /// When using multiple logging sources, exclude logs with a `flutter:` prefix
+  /// if they have already been added to the stream. This is to prevent duplicates
+  /// from being printed.
   bool _excludeLog(String message, IOSDeviceLogSource source) {
-    if (!useBothLogDeviceReaders) {
+    if (!usingMultipleLoggingSources) {
       return false;
     }
     if (message.startsWith('flutter:')) {
       if (_streamFlutterMessages.contains(message)) {
         return true;
       }
       _streamFlutterMessages.add(message);
-    } else if (source == IOSDeviceLogSource.idevicesyslog) {
+    } else if (useIOSDeployLogging && source == IOSDeviceLogSource.idevicesyslog) {
```

Comment:
why is ios-deploy use a boolean and idevicesyslog checks the source? Can it be `&& useSyslogLogging`?

Reply:
It's explained in comment below. When using `ios-deploy` and `idevicesyslog` simultaneously to get logs, exclude logs from `idevicesyslog`. So we check if we're using `ios-deploy` and we check if the message originiated from `idevicesyslog` and if it did, we exclude it.